bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/handler/GenericMQTTThingHandler.java

```diff
@@ -185,8 +186,8 @@ public void initialize() {
     }
 
     @Override
-    protected void updateThingStatus(boolean messageReceived, boolean availibilityTopicsSeen) {
-        if (messageReceived || availibilityTopicsSeen) {
+    protected void updateThingStatus(boolean messageReceived, Optional<Boolean> availibilityTopicsSeen) {
+        if (availibilityTopicsSeen.orElse(true)) {
```

Comment:
So messageReceived isn't used anymore?

Reply:
As I am thinking about it, you are right. Using the `orElse()` we can introduce the `messageReceived` also here.
There was some problem before, as the `messageReceived` flag would never be reset for a generic MQTT thing, the thing would never go ofline, even if the availability topic indicated.
But that is solved by prioritizing the availablity topic.

Reply:
@jochen314, are you going to change this? :wink:

Reply:
Ooops, I had it commited, but the push did not go through :-/

Reply:
Just noticed, that the tests are failing because of this change:
In generic MQTT thing handler, received messageReceived is not used. Using it would break existing components.
But in the homeassistent thing it is used.
So i changed it back...

Reply:
Issue #4863 needs this!